Metalama.Testing.UnitTesting/TestCompilationFactory.cs

```diff
@@ -214,6 +214,56 @@ public static CSharpCompilation CreateCSharpCompilation(
         return mainRoslynCompilation;
     }
 
+    public static SyntaxTree[] ParseCSharpSyntaxTrees( IReadOnlyDictionary<string, string> code, IEnumerable<string>? preprocessorSymbols = null )
```

Comment:
I don't think this should go into our into our testing public API, rather in the `Metalama.Framework.Tests.UnitTests` project itself.